### ISSUE TITLE
Adding Python API for GCP

### ISSUE THREAD

**Issue by corinwagen** [pyscf/pyscf#2044]
Hey all, 

Since active development on [GCP](https://github.com/grimme-lab/gcp) from Grimme's group seems minimal these days, it seems like any attempt to integrate 3c composite methods into PySCF will require custom Python bindings. This has been mentioned a few times previously: #1358, #825 

I wrote Python bindings for GCP for another project that wraps the parent ``gcp_call`` function and takes numpy arrays, but I've never contributed to PySCF before - so I'm not sure how to integrate this best, how to link to presumably conda-installed gcp, and so forth.

Would this be a useful feature for me to work on? If so, any guidance would be appreciated.

**Comment by hebrewsnabla** [comment]
Yes, it's useful. You can follow [simple-dftd3 python api](https://github.com/dftd3/simple-dftd3/tree/main/python#pyscf-support) , make it callable by pyscf and post it in your repository first.

**Comment by sunqm** [comment]
Related to https://github.com/pyscf/dispersion/pull/2
